### ISSUE TITLE
Upload purge tests fail in Go 1.5

### ISSUE THREAD

**Issue by stevvooe** [docker/distribution#887]
While we haven't yet upgrade to Go 1.5, I've been developing with it locally. While running the storage tests, I encountered the following test failure:

```console
$ go test -run 'TestPurge.*'
--- FAIL: TestPurgeGather (0.00s)
	purgeuploads_test.go:54: Unexpected upload file count: 5 != 0
INFO[0000] PurgeUploads starting: olderThan=2015-08-17 16:48:14.3179099 -0700 PDT, actuallyDelete=true
INFO[0000] Purge uploads finished.  Num deleted=0, num errors=0
INFO[0000] PurgeUploads starting: olderThan=2015-08-17 17:48:14.319165074 -0700 PDT, actuallyDelete=true
INFO[0000] Purge uploads finished.  Num deleted=0, num errors=0
--- FAIL: TestPurgeAll (0.00s)
	purgeuploads_test.go:86: Unexpectedly deleted file count 0 != 11
INFO[0000] PurgeUploads starting: olderThan=2015-08-17 17:48:14.320513555 -0700 PDT, actuallyDelete=true
INFO[0000] Purge uploads finished.  Num deleted=0, num errors=0
--- FAIL: TestPurgeSome (0.00s)
	purgeuploads_test.go:107: Unexpectedly deleted file count 0 != 5
INFO[0000] PurgeUploads starting: olderThan=2015-08-17 17:48:14.321813172 -0700 PDT, actuallyDelete=true
INFO[0000] Purge uploads finished.  Num deleted=0, num errors=0
INFO[0000] PurgeUploads starting: olderThan=2015-08-17 17:48:14.322495449 -0700 PDT, actuallyDelete=true
INFO[0000] Purge uploads finished.  Num deleted=0, num errors=0
FAIL
exit status 1
FAIL	github.com/docker/distribution/registry/storage	0.018s
```

I confirmed this was caused by 1.5 by switching back to 1.4.2. The unit tests passed.

 cc @RichardScothern

**Comment by mssola** [comment]
I've tested this with both Go 1.5 and current master and tests are passing for me. Can you still reproduce this ?

**Comment by RichardScothern** [comment]
Yep, I have a handle on this.  Fix forthcoming..

**Comment by RichardScothern** [comment]
Issue was in go 1.5rc-1 . It has been fixed for 1.5
